Makefile

```diff
@@ -0,0 +1,80 @@
+.DEFAULT_GOAL := help
+
+SHELL := /bin/bash -euo pipefail
+
+REPO_ROOT := $(CURDIR)
+
+INTERACTIVE := $(shell [ -t 0 ] && echo 1)
+
+ifneq ($(wildcard $(REPO_ROOT)/.pre-commit-config.yaml),)
+	PRE_COMMIT_CONFIG_FILE ?= $(REPO_ROOT)/.pre-commit-config.yaml
+else
+	PRE_COMMIT_CONFIG_FILE ?= $(REPO_ROOT)/repo-infra/.pre-commit-config.yaml
+endif
+
+.PHONY: clean
+clean: ## remove files created during build
+	$(call print-target)
+	cd tests && rm -f coverage.*
+
+.PHONY: install-tools
+install-tools: ## go install tools
+	$(call print-target)
+	cd tools && go install -v $(shell cd tools && go list -f '{{ join .Imports " " }}' -tags=tools)
+
+.PHONY: lint
+lint: ## golangci-lint
+lint: install-tools
+	$(call print-target)
+	cd tests && golangci-lint run -c ${REPO_ROOT}/.golangci.yml --fix
+
+.PHONY: test
+test: ## go test with race detector and code coverage
+test: install-tools
```

Comment:
need to install gotestsum as a dep. see https://github.com/mesosphere/kommander/blob/main/make/tools.mk#L71-L72

Reply:
I took the `install-tools` target from https://github.com/mesosphere/golang-repository-template, which ends up doing something similar but installs all the tools at once. I did have to fix it in this commit 78b3b5d26f2668cf4bc84466fc2db814317ffbc3 to fix the gotestsum issue i think

```
cd tools && go install -v $(shell cd tools && go list -f '{{ join .Imports " " }}' -tags=tools)
```

Reply:
resolving, can revisit in follow-up pr if needed